### ISSUE TITLE
Keep getting /bin/sh: tesseract not found 

### ISSUE THREAD

**Issue by Haasie** [MohrJonas/obsidian-ocr#4]
Hello,

First of all, thank you for making and maintaining this application. I believe it could really add some value to the obsidian community!

I tried to install the OCR plugin on MacOS but keep getting this error:
![image](https://user-images.githubusercontent.com/45860400/191239419-ab709f9a-591c-48e9-9ecc-6f2bbbaa2b52.png)

I added these values to /private/etc/paths/
![image](https://user-images.githubusercontent.com/45860400/191239580-d5cd377b-d142-4ad1-9dae-8c9ac88d5f0b.png)

I don not know how to properly add the paths to my .zshrc file. Is there something else i'am missing?

**Comment by MohrJonas** [comment]
Unfortunately I have no idea how MacOS handles paths so I can't really help you.
However, according to this StackOverflow answer https://stackoverflow.com/a/52579908, the path is set in a file called `.profile`.
Maybe @kphcs can help, he provided the instructions for MacOS.

**Comment by Haasie** [comment]
Thank you for your comment. 

I added ```usr/local/Cellar/tesseract/5.2.0/bin/``` to mu .zshrc and after that i executed ```source .zshrc```
After that i still got the pop-up on start up with tree red crosses. 

This is the outcome from ```echo $PATH```:
```/opt/homebrew/bin:/usr/local/sbin:/opt/homebrew/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Applications/VMware Fusion.app/Contents/Public:/Library/Apple/usr/bin:/Applications/Wireshark.app/Contents/MacOS:/Applications/Postgres.app/Contents/Versions/latest/bin:/usr/local/opt/fzf/bin:/usr/local/Cellar/tesseract/5.2.0/bin/```

I am are that i still mis the others but first want to try to get a green V for tesseract.

**Comment by esodesod** [comment]
Hi!

Running MacOS Monterey (12.6), on M1/Apple SoC and I (also) have all dependencies installed via Homebrew. Today, I got the tesseract provider working (yey), but only as a manual workaround, which needs to be (re)applied in Obsidian "Console" every time Obsidian restarts (for now). Hopefully my notes may help figuring out how to improve the plugin :-).

BTW: I'm no programmer by any means, but I love both troubleshooting (and OCR), so I gave it little troubleshooting session today, for fun.

As mentioned, I got all dependencies installed using Homebrew. Installing packages using homebrew, usually results in all executables are available (as symlinks) under `/opt/homebrew/bin`, and the symlinks are automatically maintained via Homebrew (e.g. new versions, updated symlinks, etc.).

The path `/opt/homebrew/bin` is usually included in path (making all executables available); at least in terminal/shell/zsh, etc. (and yes; the path may be set/updated in multiple ways, as kindly provided in the README), but Obsidian does not seem to pick this up, at least from the testing I did below. I don't know if this is Electron specific (e.g. "Electron variables"), or some other variable that may be updated, but I found a way to manually update it, after Obsidian is running. The downside (for now), is that it's not persistent, meaning restarting Obsidian (fully), the workaround needs to be re-applied.

## The workaround
Using the developer tools in Obsidian (opt+cmd+i on MacOS), and using the "Console", I checked current path variable, using the command below (thank stackoverflow for that one).
```text
console.log(process.env.PATH)
/usr/bin:/bin:/usr/sbin:/sbin
```

I then tried to update the variable, adding the /opt/homebrew/bin (makes all homebrew executables available in the path variable)
```text
process.env.PATH="/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin"
'/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin'
```

I then reloaded Obsidian, using command **reload app without saving** (so only reload, I did NOT close Obsidian fully, as the variable then would be lost again).

Now, I was **successfully able able to select the OCR Provider "Tesseract"**, from the Obsidian OCR plugin settings (whereas **previously** it always failed with Provider *"Tesseract" is not usable because: "tesseract wasn't found"*)

I also verified that the OCR engine was working as expected, by pasting in some new images, and noticed that the OCR was available (and I can now successfully search the OCR-data).

For now, this will only be a manually workaround. As said, I'm no programmer, and don't know how to make the variable persistent after restarting Obsidian. Maybe the path `/opt/homebrew/bin` can be added in the plugin somehow(?), or at least the path to the executables needed (tesserat, gm, magick, gs, etc.). If possible, I suggest using the **symlinks**, as the full path to the executables will change, depending on which versions are installed.

Verifying symlinks to executables availabe under `/opt/homebrew/bin`
```bash
which tesseract gm magick gs
/opt/homebrew/bin/tesseract
/opt/homebrew/bin/gm
/opt/homebrew/bin/magick
/opt/homebrew/bin/gs
```

Verifying my symlinks and path to the executables
```bash
for i in $(which tesseract gm magick gs);do ls -1l $i;done
lrwxr-xr-x 1 esod admin 39 Jul 15 12:13 /opt/homebrew/bin/tesseract -> ../Cellar/tesseract/5.2.0/bin/tesseract*
lrwxr-xr-x 1 esod admin 38 Jul 16 19:01 /opt/homebrew/bin/gm -> ../Cellar/graphicsmagick/1.3.38/bin/gm*
lrwxr-xr-x 1 esod admin 41 Jul 23 10:00 /opt/homebrew/bin/magick -> ../Cellar/imagemagick/7.1.0-43/bin/magick*
lrwxr-xr-x 1 esod admin 35 Jul 16 19:00 /opt/homebrew/bin/gs -> ../Cellar/ghostscript/9.56.1/bin/gs*
```

**Comment by MohrJonas** [comment]
Thank you, @esodesod, for your troubleshooting efforts.
As a workaround for this issue, the plugin could, if the path `/opt/homebrew/bin` exist, add it automatically to obsidian's path.

**Comment by MohrJonas** [comment]
Should (hopefully) be fixed as of release [1.3.2](https://github.com/MohrJonas/obsidian-ocr/releases/tag/1.3.2)
Please let me know if it worked

**Comment by esodesod** [comment]
Just verfied it's working (as expected) in version 1.3.2, on my MacOS - also after closing Obsidian (this time). The path now includes `/opt/homebrew/bin`, which makes it **detect tesseract succesfully** (and other executables within the same path, as well). Thanks!

**Comment by MohrJonas** [comment]
You're very welcome.
This fix wouldn't have been possible without your troubleshooting efforts 😊

**Comment by esodesod** [comment]
Glad it helped! OCR is such a lovely addition in Obsidian, and having the OCR data easily accessible within Obsidian is really **awesome**. So thank you, for all your time developing this plugin. 🙏

**Comment by josephdong1000** [comment]
Just wanted to put in my 2 cents, I'm using a mac but my binaries were located in `/usr/local/bin/` (I think this is because of some setting I changed with homebrew way back when, probably because I was installing some research code). I was able to get it to work by adding in another if conditional in the main.js file which wasn't too bad, though I'm not sure if adding that line would work for everyone because of weird $path search orders and whatnot.

But thanks for making this plugin! Works like a charm and greatly enhances my ability to search for stuff that I forgot about in my vault!
